backend/baseclass.pm

```diff
@@ -860,8 +860,7 @@ sub wait_serial ($self, $args) {
     my $matched = 0;
     my $str;
 
-    confess '\'current_console\' is not set' unless $self->{current_console};
```

Comment:
Because the current_console is only used in the if block below which only applies to serial_terminal but not to the regular serial0.txt case.
There is no reason to require a console other than for this if block. And as you see, I added the check to the if clause itself. Looking at the related issue that you referenced in your commit description, we shouldn't have a regression with this new approach, as the check remains.

No need to break other use cases for that.

Reply:
yep, ok. Thank you.